### DISCUSSION TITLE
(問題、變數型態) LIOJ1026=判斷等比數列

### DISCUSSION THREAD

**Discussion by s103071049** [Lidemy/forum#70] [General]
嗨以，大家好~
我自己寫的時候會用Number( )將每一個component轉成數字，再用數字去做除法運算
課程影片中，是直接將string做除法運算
我對string做四則運算後，發現除了加法會是兩個string加在一起，其他情況同數字的運算
這是 js，string為數字時特定的用法，對嗎?
```
let a =['1','3','9']
console.log(a[2]/a[1]) // 3
console.log(a[2]+a[1]) // 93
console.log(a[2]*a[1]) //27
console.log(a[2]-a[1]-a[0]) //5
console.log(typeof(a[2]),typeof(a[1])) //string string

```

**Comment by aszx87410** [comment]
JS 在使用算術運算的時候，除了加法之外，乘法除法減法都會先嘗試把東西轉成數字
如果沒辦法轉就會回一個 NaN，Not a Number 的意思

所以沒錯，可以直接把可以轉成數字的字串做除法運算

**Reply by s103071049** [reply]
了解，老師謝謝
